.github/actions/upload-sbom/action.yaml

```diff
@@ -0,0 +1,31 @@
+name: Upload SBOM
+description: uploads a sbom for a given image to a registry
+inputs:
+  image:
+    description: full image tag that the sbom will be generated for
+    required: true
+  sbom:
+    description: the sbom to attach
+    required: true
+  signing-key:
+    description: private signing key
+    required: true
+  signing-password:
+    description: password for private signing key
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Install Cosign
+      uses: sigstore/cosign-installer@dd6b2e2b610a11fd73dd187a43d57cc1394e35f9 # v3.0.5
+    - name: Attach sbom to image
+      shell: bash
+      run: |
+        cosign attach sbom --sbom "${{ inputs.sbom }}" "${{ inputs.image }}" --input-format json --type cyclonedx
+    - name: Attach sbom attestation to image
+      shell: bash
+      run: |
+        cosign attest -y --key env://COSIGN_PRIVATE_KEY  --type cyclonedx --predicate ${{ inputs.sbom }} ${{ inputs.image }}
```

Comment:
does this mean the sbom is only signed with the private key and not keyless in this case?

Reply:
yes, i just tested it if i attest it both ways (with key and keyless) it works aswell, will add keyless way aswell